src/foraging_gui/MyFunctions.py

```diff
@@ -890,7 +897,56 @@ def _LickSta(self,Trials=None):
             self.Start_CoCue_LeftRightRatio=np.nan
         else:
             self.Start_CoCue_LeftRightRatio=np.array(sum(self.Start_GoCue_LeftLicks))/np.array(sum(self.Start_GoCue_RightLicks))
-        
+
+    def calculate_inter_lick_intervals(self):
+        """
+        Calculate and categorize lick intervals
+        """
+
+        right = self.B_RightLickTime
+        left = self.B_LeftLickTime
+        threshold = .1
+
+        if right is not None and left is not None and len(right) > 0 and len(left) > 0:
+            # calculate left interval and fraction
+            same_side_l = np.diff(left)
+            same_side_l_frac = len(same_side_l[~(same_side_l > threshold)]) / len(same_side_l)
```

Comment:
On line 913 you can considering clarify this code:
```
same_side_l_frac = len(same_side_l[~(same_side_l > threshold)]) / len(same_side_l)
```
into this
```
same_side_l_frac = np.mean(same_side_l  <= threshold)
```

Notes:
- `~(same_side_l > threshold)` is the same as `same_side_l <= threshold`
- You can get the count of a boolean array with `np.sum(same_side_l <=threshold)`, and then divide by `len(same_side_l)`
- You can get the % true of a boolean array with `np.mean(same_side_l <= threshold)`